### PR TITLE
Feat: Basic languages page

### PR DESCRIPTION
**Pull request by mledl**
This PR adds the basic languages page including the following features:

- LanguageSelect with available languages
- LanguageTable with add, delete & upsert functionality including corresponding CRUD
- Corresponding Routing

What is missing and will be done in a subsequent PR:

- E2E Testing
- ~~Fallback language input field as select with available options~~
- ~~Confirmation Modal before deleting a language~~

### CODE REVIEW

File: src/components/container/language/schema.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { z } from 'zod'
+import { type LanguageCode, availableLanguages } from './languages'
+
+export const languageSchema = z.object({
+	id: z.number().optional(),
+	code: z.enum(Object.keys(availableLanguages) as [LanguageCode, ...LanguageCode[]], {
```
Comment (sjaghori) [inline src/components/container/language/schema.ts:6]: we need to give the user the ability to freely update a language code. Currently it is not possible to have language code with 'de':

image

File: src/components/container/language/languages.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+	'ar-lb': 'Arabic (Lebanon)',
+	'ar-ly': 'Arabic (Libya)',
+	'ar-ma': 'Arabic (Morocco)',
+	'ar-om': 'Arabic (Oman)',
+	'ar-qa': 'Arabic (Qatar)',
+	'ar-sa': 'Arabic (Saudi Arabia)',
+	'ar-sy': 'Arabic (Syria)',
+	'ar-tn': 'Arabic (Tunisia)',
+	'ar-ae': 'Arabic (U.A.E.)',
+	'ar-ye': 'Arabic (Yemen)',
```
Comment (sjaghori) [inline src/components/container/language/languages.ts:20]: let's just go with arabic and not the localisation. We should give the user the power to add a custom localisation.

**Review by sjaghori — CHANGES_REQUESTED** [review]
Thank you for the PR 🚀
There are some minor issues with the validation, woud you like to have a look at that?

File: src/components/container/language/schema.ts (hunk 1 above)
Comment (sjaghori) [reply src/components/container/language/schema.ts:6]: updating existing languages create new language in the backend. When I refresh the page, then I see the older entry

File: src/components/container/language/languages.ts (hunk 2 above)
Comment (mledl) [reply src/components/container/language/languages.ts:20]: Let's let them in, but we will allow the user to adapt the locale as well as you suggested in the above comment.

File: src/components/container/language/schema.ts (hunk 1 above)
Comment (mledl) [reply src/components/container/language/schema.ts:6]: Did you forget to click the save button? I tested it again and with clicking save the row gets updated correctly.

File: src/components/container/language/languages.ts [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+	'en-gb': 'English (United Kingdom)',
+	'en-us': 'English (United States)',
+	'en-zw': 'English (Zimbabwe)',
+	eo: 'Esperanto',
+	et: 'Estonian',
+	fo: 'Faeroese',
+	fa: 'Farsi',
+	fj: 'Fijian',
+	fi: 'Finnish',
+	'fr-be': 'French (Belgium)',
```
Comment (sjaghori) [inline src/components/container/language/languages.ts:66]: Please also just add (fr - French), (de - German), (es - Spanish), some users might get confused when they only want to specify the language and not the country.

**Review by sjaghori — APPROVED** [review]
Great job! :rocket:  I think we can merge this already.
There are just three things we should discuss:
1. IMO it makes more sense to automatically select the fallback language when a user adds a new language. It is perfectly fine to give them the option to deselect the fallback language or choose another fallback language.
2. We should consider renaming base language to default language
3. Adding a modal when there are unsaved changes or removing the save option, adding the functionality for what you see is what is in the db.

**Comment by sjaghori** [comment]
Oh and please add a little margin of 12 to 16px between the selector and the table:

Screenshot from 2024-09-28 12-29-10

**Comment by mledl** [comment]
> Great job! 🚀 I think we can merge this already. There are just three things we should discuss:
> 
> 1. IMO it makes more sense to automatically select the fallback language when a user adds a new language. It is perfectly fine to give them the option to deselect the fallback language or choose another fallback language.
> 2. We should consider renaming base language to default language
> 3. Adding a modal when there are unsaved changes or removing the save option, adding the functionality for what you see is what is in the db.

Will create a follow up PR for your third point. Will introduce both, improved save button state as well as the modal if there are stale changes. Moreover, I will set a the base language as fallback after creating a new language.

**Review by sjaghori — APPROVED** [review]

**Pull request merged**